UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/HealthCheck/FileShareServiceHealthClient.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Options;
+using System.Diagnostics.CodeAnalysis;
+using UKHO.FileShareClient;
+using UKHO.FileShareClient.Models;
+using UKHO.MaritimeSafetyInformation.Common.Configuration;
+using UKHO.MaritimeSafetyInformation.Common.Helpers;
+
+namespace UKHO.MaritimeSafetyInformation.Common.HealthCheck
+{
+    [ExcludeFromCodeCoverage]
+    public class FileShareServiceHealthClient : IFileShareServiceHealthClient
+    {
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly IOptions<FileShareServiceConfiguration> _fileShareServiceConfig;
+        private readonly IAuthFssTokenProvider _authFssTokenProvider;
+
+        public FileShareServiceHealthClient(IHttpClientFactory httpClientFactory, IOptions<FileShareServiceConfiguration> fileShareServiceConfig, IAuthFssTokenProvider authFssTokenProvider)
+        {
+            _httpClientFactory = httpClientFactory;
+            _fileShareServiceConfig = fileShareServiceConfig;
+            _authFssTokenProvider = authFssTokenProvider;
+        }
+
+        public async Task<HealthCheckResult> CheckHealthAsync()
+        {
+            try
+            {
+                string accessToken = await _authFssTokenProvider.GenerateADAccessToken(Guid.NewGuid().ToString());
+                FileShareApiClient fileShareApiClient = new(_httpClientFactory, _fileShareServiceConfig.Value.BaseUrl, accessToken);
+
+                IResult<BatchSearchResponse> result = await FSSSearchAsync(fileShareApiClient);
+
+                if (result.IsSuccess)
+                {
+                    return HealthCheckResult.Healthy("File share service is healthy");
```

Comment:
Please change  File share service to File Share Service at all places.
```suggestion
                    return HealthCheckResult.Healthy("File Share Service is healthy");
```

Reply:
Changes done.